### DISCUSSION TITLE
SSL Error when Accessing DataLake via Java Driver

### DISCUSSION THREAD

**Discussion by lleonardx** [dbeaver/dbeaver#35776] [Q&A] [answered]
### Description

**Issue Description:** 
I am experiencing an SSL error when trying to access the Data Lake through a connection via driver and Java library. The system cannot establish a secure connection due to a problem with verifying the SSL certificate chain. Below is the detailed error message in attach file.

**Impact:**	
It is not possible to access the Data Lake via the Java connection.	
The system fails to establish a secure connection with the server, directly affecting operations dependent on the Data Lake.

**Requested Action:**	
Please verify the validity and configuration of the certificates on the Data Lake server accessed via the Java driver.
Guidance is needed to fix the certificate chain if required, or to update the keystore if the issue is on the client side.

### DBeaver Version

Community Edition 24.2.1

### Operating System

Windows 11

### Database and driver

Infor Data Lake

Driver: infor-compass-jdbc-2020-09.jar

**Comment by E1izabeth** [comment]
Please send us log files. You can collect them to archive automatically with `Help->Collect Diagnostic Info` or manually following this guide: https://dbeaver.com/docs/wiki/Log-files/. Thank you.
![image](https://github.com/user-attachments/assets/51c7ca2e-6be2-489e-b495-028b9aa0e386)

**Comment by E1izabeth** [comment, marked as answer]
Please try changing the setting `Use Windows trust store` in Preferences->Connections and restarting DBeaver. If you haven't overridden the trust store, it might help.

**Comment by lleonardx** [comment]
[Uploading dbeaver-diagnostic-info-1727794312160.zip…]()

**Comment by E1izabeth** [comment]
Something went wrong while uploading an archive. Could you try again and wait until `Uploading dbeaver-diagnostic-info-1727794312160.zip…` will disappear?

**Comment by lleonardx** [comment]
[dbeaver-diagnostic-info-1727794312160.zip](https://github.com/user-attachments/files/17228886/dbeaver-diagnostic-info-1727794312160.zip)

**Comment by lleonardx** [comment]
it's ok now

**Comment by E1izabeth** [comment]
thank you

**Comment by lleonardx** [comment]
Use `Windows trust store ` in Preferences->Connections is not available in the Community version

**Comment by E1izabeth** [comment]
The error in logs is exactly the same as in this issue https://github.com/dbeaver/dbeaver/issues/35686
The reporter said that the issue was resolved for him by providing updated TLS certs for their datalake access

**Comment by E1izabeth** [comment]
I can find this setting in Community Edition, is it disabled or missing for you?
![image](https://github.com/user-attachments/assets/f3532230-d73d-4a8c-ad2b-a7602359d7a7)

**Comment by lleonardx** [comment]
This option is enable
![image](https://github.com/user-attachments/assets/cd5e350a-76d6-4f09-ba08-f4db4f9de316)

**Comment by lleonardx** [comment]
I'm going to test again and restart DBeaver, and test the connection with datalake.

**Comment by E1izabeth** [comment]
I see, could you try disabling it? Maybe it's vise versa and Java certs will help

**Comment by lleonardx** [comment]
When i test disable the `Windows trust store` option and restart DBeaver, the connection with DataLake has been reestablished.
Thanks for helping me!
